lib/clientv2/bgpconfig.go

```diff
@@ -0,0 +1,145 @@
+	if out != nil {
+		return out.(*apiv2.BGPConfiguration), err
+	}
+	return nil, err
+}
+
+// Delete takes name of the BGPConfiguration and deletes it. Returns an
+// error if one occurs.
+func (r bgpConfigurations) Delete(ctx context.Context, name string, opts options.DeleteOptions) (*apiv2.BGPConfiguration, error) {
+	// Check if we are trying to delete "default". Prevent deletion for now.
+	if name == "default" {
+		return nil, errors.New("Cannot delete default BGP Configuration.")
+	}
+	out, err := r.client.resources.Delete(ctx, opts, apiv2.KindBGPConfiguration, noNamespace, name)
+	if out != nil {
+		return out.(*apiv2.BGPConfiguration), err
+	}
+	return nil, err
+}
+
+// Get takes name of the BGPConfiguration, and returns the corresponding
+// BGPConfiguration object, and an error if there is any.
+func (r bgpConfigurations) Get(ctx context.Context, name string, opts options.GetOptions) (*apiv2.BGPConfiguration, error) {
+	out, err := r.client.resources.Get(ctx, opts, apiv2.KindBGPConfiguration, noNamespace, name)
+	if out != nil {
+		return out.(*apiv2.BGPConfiguration), err
+	}
+	return nil, err
+}
+
+// List returns the list of BGPConfiguration objects that match the supplied options.
+func (r bgpConfigurations) List(ctx context.Context, opts options.ListOptions) (*apiv2.BGPConfigurationList, error) {
+	res := &apiv2.BGPConfigurationList{}
+	if err := r.client.resources.List(ctx, opts, apiv2.KindBGPConfiguration, apiv2.KindBGPConfigurationList, res); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+// Watch returns a watch.Interface that watches the BGPConfiguration that
+// match the supplied options.
+func (r bgpConfigurations) Watch(ctx context.Context, opts options.ListOptions) (watch.Interface, error) {
+	return r.client.resources.Watch(ctx, opts, apiv2.KindBGPConfiguration)
+}
+
+func (r bgpConfigurations) ValidateDefaultOnlyFields(res *apiv2.BGPConfiguration) error {
+	var err error
+	if res.ObjectMeta.GetName() != "default" {
+		if res.Spec.NodeToNodeMeshEnabled != nil {
+			err = cerrors.ErrorValidation{
+				ErroredFields: []cerrors.ErroredField{{
+					Name:   "BGPConfiguration.Spec.NodeToNodeMeshEnabled",
+					Reason: "Cannot set nodeToNodeMeshEnabled on a non default BGP Configuration.",
+				}},
+			}
+		}
+
+		if res.Spec.DefaultNodeASNumber != nil {
+			if err != nil {
+				// Need to convert the type to add to the correct fields
+				tempErr := err.(cerrors.ErrorValidation)
+				tempErr.ErroredFields = append(tempErr.ErroredFields, cerrors.ErroredField{
+					Name:   "BGPConfiguration.Spec.DefaultNodeASNumber",
+					Reason: "Cannot set defaultNodeASNumber on a non default BGP Configuration.",
+				})
+				err = tempErr
+			} else {
+				err = cerrors.ErrorValidation{
+					ErroredFields: []cerrors.ErroredField{{
+						Name:   "BGPConfiguration.Spec.DefaultNodeASNumber",
+						Reason: "Cannot set defaultNodeASNumber on a non default BGP Configuration.",
+					}},
+				}
```

Comment:
So I think this workflow would have been a little easier if you'd done something like:

```
erroredFields := []errors.ErroredField{}
if meshSet {
    erroredFields = append(erroredFields, errors.ErroredField{meshField})
}
if asNumberSet {
    erroredFields = append(erroredFields, errors.ErroredField{asNumberField})
}
if len(erroredFields) > 0 {
    return ErrorValidation{ErroredFields: erroredFields}
}
```